apps/smart-contracts/core/test/MintHook.test.ts

```diff
@@ -0,0 +1,67 @@
+import chai, { expect } from 'chai'
+import { ethers } from 'hardhat'
+import { SignerWithAddress } from '@nomiclabs/hardhat-ethers/dist/src/signer-with-address'
+import { id, parseEther } from 'ethers/lib/utils'
+import { Contract } from 'ethers'
+import { MockContract, smock } from '@defi-wonderland/smock'
+import { ZERO_ADDRESS } from 'prepo-constants'
+import { mintHookFixture, smockAccountListFixture } from './fixtures/HookFixture'
+import { MintHook } from '../typechain'
+
+chai.use(smock.matchers)
+
+describe('=> MintHook', () => {
+  let deployer: SignerWithAddress
+  let user: SignerWithAddress
+  let market: SignerWithAddress
+  let mintHook: MintHook
+  let allowlist: MockContract<Contract>
+
+  beforeEach(async () => {
+    ;[deployer, user, market] = await ethers.getSigners()
+    mintHook = await mintHookFixture()
+    allowlist = await smockAccountListFixture()
+  })
+
+  describe('initial state', () => {
+    it('sets deployer to owner', async () => {
+      expect(await mintHook.owner()).to.eq(deployer.address)
+    })
+
+    it('sets nominee to zero address', async () => {
+      expect(await mintHook.getNominee()).to.eq(ZERO_ADDRESS)
+    })
+  })
+
+  describe('setAllowlist', () => {
```

Comment:
nit: (same for others)
```suggestion
  describe('# setAllowlist', () => {
```